builtin/providers/google/resource_compute_router_peer.go

```diff
@@ -0,0 +1,290 @@
+			d.SetId("")
+
+			return nil
+		}
+
+		return fmt.Errorf("Error Reading router %s/%s: %s", region, routerName, err)
+	}
+
+	var peerFound bool = false
+
+	var peers []*compute.RouterBgpPeer = router.BgpPeers
+	for _, peer := range peers {
+
+		if peer.Name == peerName {
+			peerFound = true
+			d.SetId(fmt.Sprintf("%s/%s/%s", region, routerName, peerName))
+			d.Set("interface", peer.InterfaceName)
+			d.Set("ip_address", peer.PeerIpAddress)
+			d.Set("asn", peer.PeerAsn)
+		}
+	}
+	if !peerFound {
+		log.Printf("[WARN] Removing router peer %s/%s/%s because it is gone", region, routerName, peerName)
+		d.SetId("")
+	}
+
+	return nil
+}
+
+func resourceComputeRouterPeerDelete(d *schema.ResourceData, meta interface{}) error {
+
+	config := meta.(*Config)
+
+	region, err := getRegion(d, config)
+	if err != nil {
+		return err
+	}
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	routerName := d.Get("router").(string)
+	peerName := d.Get("name").(string)
+
+	routerId := fmt.Sprintf("router/%s/%s", region, routerName)
+	mutexKV.Lock(routerId)
+	defer mutexKV.Unlock(routerId)
+
+	routersService := compute.NewRoutersService(config.clientCompute)
+	router, err := routersService.Get(project, region, routerName).Do()
+	if err != nil {
+		if gerr, ok := err.(*googleapi.Error); ok && gerr.Code == 404 {
+			log.Printf("[WARN] Removing router peer because its router %d is gone", d.Get("router").(string))
+
+			return nil
+		}
+
+		return fmt.Errorf("Error Reading Router %s: %s", routerName, err)
+	}
+
+	var peerFound bool = false
+
+	var oldIfaces []*compute.RouterBgpPeer = router.BgpPeers
+	var newIfaces []*compute.RouterBgpPeer = make([]*compute.RouterBgpPeer, len(router.BgpPeers))
+	for _, peer := range oldIfaces {
+
+		if peer.Name == peerName {
+			peerFound = true
+			continue
+		} else {
+			newIfaces = append(newIfaces, peer)
+		}
+	}
+
+	if peerFound {
+
+		log.Printf(
+			"[INFO] Removing peer %s", peerName)
+		patchRouter := &compute.Router{
+			BgpPeers: newIfaces,
+		}
+
+		log.Printf("[DEBUG] Updating router %s/%s with peers: %+v", region, routerName, newIfaces)
+		op, err := routersService.Patch(project, region, router.Name, patchRouter).Do()
+		if err != nil {
+			return fmt.Errorf("Error patching router %s/%s: %s", region, routerName, err)
+		}
+
+		err = computeOperationWaitRegion(config, op, project, region, "Patching router")
+		if err != nil {
+			return fmt.Errorf("Error waiting to patch router %s/%s: %s", region, routerName, err)
+		}
+
+	} else {
+		log.Printf("[DEBUG] Router %s/%s had no peer %s already", region, routerName, peerName)
+	}
+
+	return nil
+}
+
+func resourceComputeRouterPeerImportState(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	parts := strings.Split(d.Id(), "/")
+	if len(parts) != 3 {
+		return nil, fmt.Errorf("Invalid router specifier. Expecting {region}/{router}")
```

Comment:
I wanted to say that we expect the user to define it as <region_name>/<router_name>. How about:
"Invalid router specifier. Please define it as <region_name>/<router_name>"